### DISCUSSION TITLE
FCM - Add easy way to get all subscribed topics or simply unsubscribe from all topics.

### DISCUSSION THREAD

**Discussion by safield** [firebase/flutterfire#6563] [Feature request]
Currently, the only way for me to unsubscribe from all subscribed topics using FCM in my flutter app, is to locally store all the previously subscribed topics. This is not ideal as it is duplicating data, and could result in bugs where my local storage of subscribed topics goes out of sync with the actual subscribed topics.

FlutterFire fcm library should expose a method _getSubscribedTopics()_ which returns all of the subscribed topics. Or alternatively, provide a method called _unsubscribeFromAllTopics_, though this would be less flexible.

**Comment by Pr47h4m** [comment]
Right @safield 

I'm looking for similar functionality and I feel this can reduce programming efforts (frontend and backend) for sending notifications in multi tenant applications

**Comment by GiddyNaya** [comment]
For me I just regenerate the token if I need to unsubscribe from all topics. Yes, it's a long walk for such a simple task but does the job:

```
Future<String?> subscribeToTopic(String topic, {bool clearAll = false}) async {
  String? token;
  if(clearAll){
      await FirebaseMessaging.instance.deleteToken();
      token = await FirebaseMessaging.instance.getToken();
  }
  await FirebaseMessaging.instance.subscribeToTopic(topic);
  return token;
}

// To use...
var newToken = await subscribeToTopic("finance", clearAll:true);
if(newToken != null){
  // send to server
}
```
